tests/test_modeling_common.py

```diff
@@ -1595,6 +1597,25 @@ def test_can_use_safetensors(self):
                         f"The shared pointers are incorrect, found different pointers for keys {shared_names}",
                     )
 
+    def test_load_save_without_tied_weights(self):
```

Comment:
This new test checks that when weights are untied, they are properly saved and we complain if they are missing from the checkpoint.